lib/Sema/CSSimplify.cpp

```diff
@@ -2598,6 +2598,16 @@ bool ConstraintSystem::repairFailures(
         elt.castTo<LocatorPathElt::ApplyArgToParam>().getParamIdx() == 1)
       break;
 
+    if (auto *fix = ExpandArrayIntoVarargs::attempt(*this, lhs, rhs, locator)) {
+      conversionsOrFixes.push_back(fix);
+      // The ExpandArrayIntoVarargs fix might fail when we try to simplify it if
+      // the array element and variadic parameter types are unrelated. Fall back
+      // to a regular argument mismatch in that case.
```

Comment:
A way to address this problem might be to test whether element type of an array matches type of the parameter using matchTypes with TMF_ApplyingFix set as part of an attempt, that also helps to infer generic arguments. There are a couple of examples of how that could be done in repairFailures already.